packages/fiori/src/Timeline.ts

```diff
@@ -111,6 +188,116 @@ class Timeline extends UI5Element {
 			: Timeline.i18nBundle.getText(TIMELINE_ARIA_LABEL);
 	}
 
+	get growsOnScroll(): boolean {
+		return this.growing === TimeLineGrowingMode.Scroll;
+	}
+
+	get timeLineEndDOM(): Element {
+		return this.shadowRoot!.querySelector(".ui5-time-line-end-marker")!;
+	}
+
+	get growingButtonIcon() {
+		return this.layout === TimelineLayout.Horizontal ? "process" : "drill-down";
+	}
+
+	get moreBtn(): HTMLElement | null {
+		const domRef = this.getDomRef();
+
+		if (this.growsWithButton && domRef) {
+			return domRef.querySelector<HTMLElement>(`#${this._id}-growingButton`);
+		}
+
+		return null;
+	}
+
+	get showBusyIndicatorOverlay() {
+		return !this.growsWithButton && this.loading;
+	}
+
+	get growsWithButton(): boolean {
+		return this.growing === TimeLineGrowingMode.Button;
+	}
+
+	onAfterRendering() {
+		if (this.growsOnScroll) {
+			this.observeTimeLineEnd();
+		}
+	}
+
+	onEnterDOM() {
+		this.growingIntersectionObserver = this.getIntersectionObserver();
+	}
+
+	onExitDOM() {
+		this.growingIntersectionObserver!.disconnect();
+		this.growingIntersectionObserver = null;
+		this.timeLineEndObserved = false;
+	}
+
+	observeTimeLineEnd() {
+		if (!this.timeLineEndObserved) {
+			this.getIntersectionObserver().observe(this.timeLineEndDOM);
+			this.timeLineEndObserved = true;
+		}
+	}
+
+	getIntersectionObserver(): IntersectionObserver {
+		if (!this.growingIntersectionObserver) {
+			this.growingIntersectionObserver = new IntersectionObserver(this.onInteresection.bind(this), {
+				root: document,
```

Comment:
This could cause observation outside of the component, I think using `this.getDomRef()` might be a better approach, but am not entirely sure about that.

@tsanislavgatev what do you think ?

Reply:
Maybe in the next iteration, we will discuss it with the colleagues from the core team or with @tsanislavgatev